sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformationExec.scala

```diff
@@ -54,208 +51,170 @@ import org.apache.spark.util.{CircularBuffer, RedirectThread, SerializableConfig
  * @param script the command that should be executed.
  * @param output the attributes that are produced by the script.
  */
-case class ScriptTransformationExec(
+case class HiveScriptTransformationExec(
     input: Seq[Expression],
     script: String,
     output: Seq[Attribute],
     child: SparkPlan,
     ioschema: HiveScriptIOSchema)
-  extends UnaryExecNode {
+  extends ScriptTransformBase {
 
   override def producedAttributes: AttributeSet = outputSet -- inputSet
 
   override def outputPartitioning: Partitioning = child.outputPartitioning
```

Comment:
We cannot move `producedAttributes` and `outputPartitioning` into the base class?

Reply:
> We cannot move `producedAttributes` and `outputPartitioning` into the base class?

Good suggestion， can  move